python/ray/serve/tests/test_handle_api.py

```diff
@@ -31,12 +31,19 @@ async def __call__(self):
             response = self._handle.remote()
             assert isinstance(response, DeploymentResponse)
             val = await response
+
+            # Check that the response can be awaited twice.
+            assert (await response) == val
+
             return val
 
     handle: DeploymentHandle = serve.run(Deployment.bind(downstream.bind()))
     assert isinstance(handle, DeploymentHandle)
     assert handle.remote().result() == "hello"
 
+    # Check that `.result()` can be called twice.
+    assert handle.remote().result() == "hello"
```

Comment:
Can we test the case where `handle.remote()` is only called once, but `.result()` is called twice on the same `DeploymentResponse` object? This specific test is calling `handle.remote()` twice, so I think there are two separate `DeploymentResponse` objects.

Reply:
fixed